src/modules/adaptors/Frax/CollateralFTokenAdaptor.sol

```diff
@@ -0,0 +1,226 @@
+     * @param adaptorData the collateral asset deposited into Fraxlend
+     * NOTE: CRISPY QUESTION - TODO: confirm that this works... EIN - FraxlendCore doesn't have an interface to access the getter. So we'll have to have it here.
+     * TODO: confirm that there is no need to typeCast on the decoded adaptorData when needing address, ERC20 or vice versa.
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        ICollateralFToken fToken = ICollateralFToken(abi.decode(adaptorData, (address))); // TODO: CRISPY QUESTION - could change adaptorData to be collateralToken but then it won't be consistent with rest of fraxlend adaptors.
+        return fToken.userCollateralBalance(msg.sender) + fToken.collateralContract().balanceOf(msg.sender); // reports balance of collateral provided to protocol.
+    }
+
+    /**
+     * @notice Returns collateral asset
+     */
+    function assetOf(bytes memory) public view override returns (ERC20) {
+        IFToken fToken = abi.decode(adaptorData, (IFToken));
+        return ERC20(fToken.collateralContract());
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to add collateral to the respective cellar position on FraxLend, enabling borrowing.
+     * @dev `borrowTokens()` within DebtFTokenAdaptor can increase cellar collateralPosition too. `borrowTokens()` cannot be used without a trusted CollateralFTokenAdaptor position in the same cellar because CollateralFTokenAdaptors track the collateral within a Fraxlend pair. CRISPY QUESTION - See TODO: within DebtFTokenAdaptor regarding this.
+     */
+    function addCollateral(
+        ERC20 _collateralToken,
+        uint256 _collateralToDeposit,
+        IFToken _fToken
+    ) public {
+        // _validateFToken(collateralToken); // TODO: CRISPY QUESTION - we could have a validation helper but Fraxlend checks if there is even a collateral position for whomever is attempting to borrow. So it is not necessary unless we want to save Strategists gas during reversions. I have left this here in case I am wrong.
+        // amountToDeposit = _maxAvailable(collateralToken, amountToDeposit); // TODO: CRISPY QUESTION - not sure if we want to deposit the max or not by default... It depends if the strategist wants to do uint256.max or not for these cellars.
+        address fraxlendPair = address(_fToken);
+        _collateralToken.safeApprove(fraxlendPair, _collateralToDeposit);
+        _fraxLendPair.addCollateral(_collateralToDeposit, address(this));
+
+        // TODO: CRISPY QUESTION - should we check that the LTV is in a healthy state? Check that the FraxlendCore doesn't already do that.
```

Comment:
This function is adding collateral, so I dont think we need to check the LTV, but on `removeCollateral` we should check the LTV.